@DQ/Q4.m

```diff
@@ -0,0 +1,61 @@
+% Q4(x) returns the partial derivative of the unit quaternion r with
+%       respect to log(r).
+%       See Eq. (22) of Savino et al (2020). 
+%       Pose consensus based on dual quaternion algebra with application to 
+%       decentralized formation control of mobile manipulators.
+%       https://doi.org/10.1016/j.jfranklin.2019.09.045
+
+% (C) Copyright 2011-2023 DQ Robotics Developers
+% 
+% This file is part of DQ Robotics.
+% 
+%     DQ Robotics is free software: you can redistribute it and/or modify
+%     it under the terms of the GNU Lesser General Public License as published by
+%     the Free Software Foundation, either version 3 of the License, or
+%     (at your option) any later version.
+% 
+%     DQ Robotics is distributed in the hope that it will be useful,
+%     but WITHOUT ANY WARRANTY; without even the implied warranty of
+%     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%     GNU Lesser General Public License for more details.
+% 
+%     You should have received a copy of the GNU Lesser General Public License
+%     along with DQ Robotics.  If not, see <http://www.gnu.org/licenses/>.
+%
+% DQ Robotics website: dqrobotics.github.io
+%
+% Contributors to this file:
+% 1. Bruno Vilhena Adorno (adorno@ieee.org)
+%        - Responsible for the original implementation.
+%
+% 2. Juan Jose Quiroz Omana (juanjqo@g.ecc.u-tokyo.ac.jp)
+%      Implemented the modifications requested in #90
+%          (https://github.com/dqrobotics/matlab/issues/90)
+%        - Removed Q4() from Q8.m, exposed the method to the user, and
+%        - added the check for unit norm.        
+
+function ret = Q4(x)
+% Return the partial derivative of the unit quaternion r with respect to
+% log(r).
+    if ~is_unit(x)
+        error('Q4 function is defined only for unit dual quaternions');
```

Comment:
@bvadorno @juanjqo 
If I may add something, maybe it would be good to add the additional condition `is_quaternion` to match the documentation and prevent users from getting confusing results.
https://github.com/dqrobotics/matlab/blob/f5f85a3b5797a86cc6be226042d84bcdd6bb0414/%40DQ/is_quaternion.m#L29

Reply:
@bvadorno @mmmarinho I added following test:

```matlab
    if ~is_unit(x) || ~is_quaternion(x)
        error('Q4 function is defined only for unit quaternions');
    end
```
Please let me know if that makes sense.